app/models/product_feed.rb

```diff
@@ -0,0 +1,32 @@
+class ProductFeed < ActiveRecord::Base
+
+  before_destroy :add_to_history
+  after_update :add_to_history
+
+  validates :title, :description, :affiliate_code, presence: true
+
+  alias_attribute :image_urls, :image_url
+  alias_attribute :aff_code, :affiliate_code
+
+  scope :visible, -> { where(deleted: false) }
+
+  def self.create_records(file_records)
```

Comment:
@Murzic the best approach I in my opinion would be to use background jobs for import file. IF it would be a big application I could add sidekiq, gem state_machine and background workers for import and validation after save for any errors in csv file.
If any errors would be found they would be added to field with name like err_msg and outputted in front-end.